node/tests/RedisClient.test.ts

```diff
@@ -365,6 +365,92 @@ describe("GlideClient", () => {
+
+            try {
+                // neither key exists
+                checkSimple(
+                    await client.copyDB(source, destination, index1, false),
+                ).toEqual(false);
+
+                // source exists, destination does not
+                expect(await client.set(source, value1)).toEqual("OK");
+                checkSimple(
+                    await client.copyDB(source, destination, index1, false),
+                ).toEqual(true);
+                expect(await client.select(index1)).toEqual("OK");
+                checkSimple(await client.get(destination)).toEqual(value1);
+
+                // new value for source key
+                expect(await client.select(index0)).toEqual("OK");
+                expect(await client.set(source, value2)).toEqual("OK");
+
+                // no REPLACE, copying to existing key on DB 0 & 1, non-existing key on DB 2
+                checkSimple(
+                    await client.copyDB(source, destination, index1, false),
+                ).toEqual(false);
+                checkSimple(
+                    await client.copyDB(source, destination, index2, false),
+                ).toEqual(true);
+
+                // new value only gets copied to DB 2
+                expect(await client.select(index1)).toEqual("OK");
+                checkSimple(await client.get(destination)).toEqual(value1);
+                expect(await client.select(index2)).toEqual("OK");
+                checkSimple(await client.get(destination)).toEqual(value2);
+
+                // both exists, with REPLACE, when value isn't the same, source always get copied to
+                // destination
+                expect(await client.select(index0)).toEqual("OK");
+                checkSimple(
+                    await client.copyDB(source, destination, index1, true),
+                ).toEqual(true);
+                expect(await client.select(index1)).toEqual("OK");
+                checkSimple(await client.get(destination)).toEqual(value2);
+            } finally {
+                // switching back to db 0
+                await client.select(index0);
+            }
+
+            // transaction tests
+            try {
+                const transaction = new Transaction();
+                transaction.select(index1);
+                transaction.set(source, value1);
+                transaction.copy(source, destination, index1, true);
+                transaction.get(destination);
+                const results = await client.exec(transaction);
+
+                if (results != null) {
+                    checkSimple(results.length).toEqual(4);
+                    checkSimple(results[2]).toEqual(true);
+                    checkSimple(results[3]).toEqual(value1);
+                }
+            } finally {
+                // switching back to db 0
+                await client.select(index0);
+            }
+
+            client.close();
+        },
+        TIMEOUT,
+    );
+
```

Comment:
Can you add a cross slot test to RedisClusterClient.test.ts?

Reply:
I did add the cross slot test in my local but it failed to pick it up in the code review. I will re-add it.